pkg/thanos/rules.go

```diff
@@ -148,7 +149,7 @@ func (o *Operator) selectRuleNamespaces(p *monitoringv1.ThanosRuler) ([]string,
 			return namespaces, errors.Wrap(err, "convert rule namespace label selector to selector")
 		}
 
-		namespaces, err = o.listMatchingNamespaces(ruleNamespaceSelector)
+		namespaces, err = prompkg.ListMatchingNamespaces(ruleNamespaceSelector, o.nsRuleInf)
```

Comment:
to avoid a dependency from pkg/thanos to pkg/prometheus, I'd move `ListMatchingNamespaces()` to pkg/operator.